--- gtk-4.10.1/modules/printbackends/meson.build.orig	2023-03-22 17:53:14.882878089 +0100
+++ gtk-4.10.1/modules/printbackends/meson.build	2023-03-22 20:30:37.471723193 +0100
@@ -26,7 +26,9 @@ if get_option('print-cpdb').enabled() an
     install_dir: printbackends_install_dir,
     install: true,
   )
-elif cups_dep.found()
+endif
+
+if not get_option('print-cups').disabled() and cups_dep.found()
   print_backends += 'cups'
   shared_module('printbackend-cups',
     sources: [
