--- gtk-4.20.3/gtk/print/backends/meson.build.orig	2025-11-20 06:08:31.000000000 +0100
+++ gtk-4.20.3/gtk/print/backends/meson.build	2025-11-27 21:24:52.845219699 +0100
@@ -14,7 +14,9 @@ if get_option('print-cpdb').enabled() an
   ])
   print_backend_deps += [cpdb_dep]
   cdata.set('HAVE_PRINTBACKEND_CPDB', 1)
-elif get_option('print-cups').enabled() and cups_dep.found()
+endif
+
+if get_option('print-cups').enabled() and cups_dep.found()
   print_backends += 'cups'
   print_backend_sources += files([
     'gtkprintbackendcups.c',
