--- gtk-4.0.0/modules/printbackends/meson.build.orig	2020-12-18 16:22:39.358636846 +0100
+++ gtk-4.0.0/modules/printbackends/meson.build	2020-12-18 16:26:30.037387152 +0100
@@ -40,7 +40,9 @@
     install_dir: printbackends_install_dir,
     install: true,
   )
-else
+endif
+
+if get_option('print-lpr') or not print_backends.contains('cups')
   # Automatic fall-back to the lpr backend
   print_backends += 'lpr'
   shared_module('printbackend-lpr',
--- gtk-3.99.4/meson_options.txt.orig	2020-11-05 03:46:55.071335300 +0100
+++ gtk-3.99.4/meson_options.txt	2020-11-08 19:19:18.240643030 +0100
@@ -39,6 +39,11 @@
 
 # Print backends
 
+option('print-lpr',
+       type: 'boolean',
+       value: false,
+       description: 'Build the lpr print backend')
+
 option('print-cups',
        type: 'feature',
        value: 'auto',
