--- gtk-4.20.3/gdk/gdkvulkancontext.c.orig	2025-11-20 06:08:31.000000000 +0100
+++ gtk-4.20.3/gdk/gdkvulkancontext.c	2025-11-29 09:50:03.396025813 +0100
@@ -688,7 +688,7 @@ gdk_vulkan_present_is_busy (GdkVulkanCon
 {
   VkResult res;
 
-  if (present->vk_swapchain == NULL)
+  if (present->vk_swapchain == VK_NULL_HANDLE)
     return FALSE;
 
   if (!present->vk_fence)
@@ -1208,7 +1208,7 @@ gdk_vulkan_context_surface_detach (GdkDr
       if (priv->presents[i].vk_swapchain)
         {
           gdk_vulkan_context_unref_swapchain (self, priv->presents[i].vk_swapchain);
-          priv->presents[i].vk_swapchain = NULL;
+          priv->presents[i].vk_swapchain = VK_NULL_HANDLE;
         }
       vkDestroySemaphore (vk_device,
                           priv->presents[i].vk_semaphore,
